### PR TITLE
Various fixes to simplify GPU build of CoreNEURON and NMODL integration with NEURON

### PR DESCRIPTION
**Pull request by iomaganaris**
- NMODL submodule updated
- Improve GPU device selection in multi-gpu environment
- Fix travis and jenkins CI
- Various CMake changes to improve GPU build
- Provide the corrent NMODL binary when NMODL is used as external package
- Fixes issue with including the `_kinderiv.h` file when used in NEURON as submodule
- Fix travis and Jenkins CI

### CODE REVIEW

**Comment by pramodk** [comment]
Please retest

File: extra/nrnivmodl_core_makefile.in [hunk 1]
```diff
@@ -115,9 +123,15 @@ $(special): $(coremech_lib)
...
-	(rm -f $(OUTPUT)/.libs/libcorenrnmech@CMAKE_SHARED_LIBRARY_SUFFIX@ ; mkdir -p $(OUTPUT)/.libs ; ln -s ../../${coremech_lib} $(OUTPUT)/.libs/libcorenrnmech@CMAKE_SHARED_LIBRARY_SUFFIX@)
+	@rm -f ${coremech_lib}
+	$(CXXCOMPILE) @CMAKE_CXX_COMPILE_OPTIONS_PIC@ -c -DADDITIONAL_MECHS $(datadir)/enginemech.cpp
+	@if [ "@COMPILE_LIBRARY_TYPE@" = "SHARED" ]; then\
+		$(CXX_LINK_SHARED) $(INCFLAGS) -I $(incdir) enginemech.o -o ${coremech_lib} ${_SONAME} \
+			$(mod_func_o) $(dimplic_o) $(mod_c_objs) $(mod_ispc_objs) $(libdir)/libscopmath.a $(CORENRNLIB_FLAGS) -Wl,-rpath,$(libdir) $(LDFLAGS);\
+	else\
+		ar cq ${coremech_lib} enginemech.o $(mod_func_o) $(dimplic_o) $(mod_c_objs) $(mod_ispc_objs);\
+	fi
+	(rm -f $(OUTPUT)/.libs/libcorenrnmech$(library_suffix) ; mkdir -p $(OUTPUT)/.libs ; ln -s ../../${coremech_lib} $(OUTPUT)/.libs/libcorenrnmech$(library_suffix))
```
Comment (pramodk) [inline extra/nrnivmodl_core_makefile.in]: I did this and looks ugly - but I don't see any easy way at the moment.

File: tests/jenkins/install_coreneuron.sh [hunk 2]
```diff
@@ -18,36 +18,35 @@ else
...
-        -DCMAKE_CXX_FLAGS:STRING="-O2 -D_GLIBCXX_USE_CXX11_ABI=0 -DR123_USE_SSE=0" \
-        -DCOMPILE_LIBRARY_TYPE=STATIC  \
-        -DCUDA_HOST_COMPILER=`which gcc` \
-        -DCUDA_PROPAGATE_HOST_FLAGS=OFF \
         -DCORENRN_ENABLE_GPU=ON \
         -DCORENRN_ENABLE_CUDA_UNIFIED_MEMORY=ON \
         -DCMAKE_INSTALL_PREFIX=$WORKSPACE/install_${CORENRN_TYPE}/ \
-        -DTEST_MPI_EXEC_BIN="srun;--exclusive;--account=proj16;--partition=interactive;--constraint=volta;--gres=gpu:1;--mem;0;-t;00:05:00" \
-        -DTEST_EXEC_PREFIX="srun;-n;6;--exclusive;--account=proj16;--partition=interactive;--constraint=volta;--gres=gpu:1;--mem;0;-t;00:05:00" \
+        -DTEST_MPI_EXEC_BIN="srun;--account=proj16;--partition=$SALLOC_PARTITION;--constraint=volta;--gres=gpu:1;--mem;0;-t;00:05:00" \
```
Comment (iomaganaris) [inline tests/jenkins/install_coreneuron.sh]: Shall we do the gpu tests with 2 or more gpus to test this feature?

File: extra/nrnivmodl_core_makefile.in (hunk 1 above)
Comment (iomaganaris) [reply extra/nrnivmodl_core_makefile.in]: I agree. We can merge the PR with that and we can improve it later if we find something better

File: tests/jenkins/install_coreneuron.sh (hunk 2 above)
Comment (pramodk) [reply tests/jenkins/install_coreneuron.sh]: yeah thats good idea. I changed it.

**Pull request merged**